sphinx_gallery/gen_gallery.py

```diff
@@ -301,3 +313,22 @@ def setup(app):
 
     app.connect('build-finished', sumarize_failing_examples)
     app.connect('build-finished', embed_code_links)
+
+
+class ExplicitOrderKey(object):
+    """Explicitly order a list of strings."""
+    def __init__(self, ordered_list):
+        if not isinstance(ordered_list, list):
+            raise ValueError('ordered_list must be a list')
+        self.ordered_list = ordered_list
+        self.unordered_items = []
+
+    def __call__(self, item):
+        if item in self.ordered_list:
```

Comment:
Unless you have a specific use case in mind, I would be in favour of requiring the user to specify all the folders and raise an exception if `item` is not in `self.ordered_list`. IIRC `os.listdir` is not deterministic so this could introduce ordering changes in the gallery that are confusing to the user.